src/codelabs/dart-cheatsheet.md

```diff
@@ -1732,59 +1728,57 @@ class Color {
   
   Color(this.red, this.green, this.blue);
 
-  // Create a named constructor called "Color.black" here:
-}
-{$ end main.dart $}
-{$ begin solution.dart $}
-class Color {
-  int red;
-  int green;
-  int blue;
-  
-  Color(this.red, this.green, this.blue);
+  // TODO: Create a named constructor called "Color.black" here:
 
-  Color.black()
-      : red = 0,
-        green = 0,
-        blue = 0;
 }
-{$ end solution.dart $}
-{$ begin test.dart $}
+
+
+// Tests your solution (Don't edit!):
 void main() {
   final errs = <String>[];
 
   try {
     final result = Color.black();
     
     if (result.red != 0) {
-      errs.add('Called Color.black() and got a Color with red equal to ${result.red} instead of the expected value (0).');
+      errs.add('Called Color.black() and got a Color with red equal to \n ${result.red} instead of the expected value (0).');
     }
 
     if (result.green != 0) {
-      errs.add('Called Color.black() and got a Color with green equal to ${result.green} instead of the expected value (0).');
+      errs.add('Called Color.black() and got a Color with green equal to \n ${result.green} instead of the expected value (0).');
     }
 
     if (result.blue != 0) {
-  errs.add('Called Color.black() and got a Color with blue equal to ${result.blue} instead of the expected value (0).');
+  errs.add('Called Color.black() and got a Color with blue equal to \n ${result.blue} instead of the expected value (0).');
     }
   } catch (e) {
-    _result(false, ['Called Color.black() and got an exception of type ${e.runtimeType}.']);
+    _result(false, ['Called Color.black() and got an exception of type \n ${e.runtimeType}.']);
     return;
   }
 
   if (errs.isEmpty) {
-    _result(true);
+    print('Success!');
   } else {
-    _result(false, errs);
+    errs.forEach(print);
   }
 }
-{$ end test.dart $}
-{$ begin hint.txt $}
-The declaration for your constructor should be `Color.black() {}`.
-Inside the braces, set red, green, and blue to zero.
-{$ end hint.txt $}
+{$ end main.dart $}
 ```
 
+<details markdown="1">
+ <summary>
+   <b>Solution for named constructors example</b>
+  </summary>
+  The declaration for your constructor should be `Color.black() {}`.
+  Inside the braces, set `red`, `green`, and `blue` to zero.
```

Comment:
We must have not updated this hint when it was changed to use an initializer list for null safety.

How does this sound?

```suggestion
  The declaration for your constructor should begin with `Color.black(): `.
  In the initializer list (after the colon), set `red`, `green`, and `blue` to `0`.
```

Reply:
Nice catch!